File: TimeTravelObjC/TimeTravelObjC/ViewController.m
//
//  ViewController.m
//  TimeTravelObjC
//
//  Created by Mitchell Phillips on 3/7/16.
//  Copyright © 2016 Wasted Potential LLC. All rights reserved.
//

#import "ViewController.h"
#import "TimeSetViewController.h"

@interface ViewController ()

@end

@implementation ViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    
       [self dates];
}
-(void)dates {
    NSDate *presentDate = [NSDate date];
    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
        [formatter setDateStyle:NSDateFormatterShortStyle];
    
    NSString *formattedDate = [formatter stringFromDate:presentDate];
    NSLog(@"presentTimeLabel%@", formattedDate);}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}
- (void)updateUI {
    
    if (count > 0) {
        count-- ;
        NSString *warpCount = [NSString stringWithFormat:@"%d", count];
        self.timeUntilWarpLabel.text = warpCount;
    } else {
        [timer invalidate];
        timer = nil;
        self.timeUntilWarpLabel.text = @"WARP!!!";
        [self performSegueWithIdentifier: @"warpSegue" sender: self];
    }
    
}
- (IBAction)unwindSegue:(UIStoryboardSegue *)segue {
    
}
- (IBAction)setDateUnwind:(UIStoryboardSegue *)segue {
    if ([segue.identifier isEqualToString:@"getTheDateSegue"]) {
       TimeSetViewController *controller = (TimeSetViewController *) segue.sourceViewController;
        newStr = controller->destinationDate;
        self.destinationTimeLabel.text = newStr;
    };
    
}



- (IBAction)setDateButton:(UIButton *)sender {
    
    
}

- (IBAction)warpButton:(UIButton *)sender {
    
    count = 10;
    
    [NSTimer scheduledTimerWithTimeInterval:0.3
                                     target:self
                                   selector:@selector(updateUI)
                                   userInfo:nil
                                    repeats:NO];
    
}
@end

File: Day 5/WarpRiders/WarpRiders/ViewController.swift
//
//  ViewController.swift
//  WarpRiders
//
//  Created by Mitchell Phillips on 2/4/16.
//  Copyright © 2016 Mitchell Phillips. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var timer: NSTimer?
    
    var count: Int = 10
    
    let speed: NSTimeInterval = 0.3
    
    
    
    @IBOutlet weak var destinationLabel: UILabel!
    
    @IBOutlet weak var presentTimeLabel: UILabel!
    
    @IBOutlet weak var lastTimeLabel: UILabel!
    
    @IBOutlet weak var timeUntilWarpLabel: UILabel!
    
    @IBAction func unwindToMainView (segue: UIStoryboardSegue) {
        
        if segue.identifier == "unwindToMainView" {
            
            let viewControllerTwo = segue.sourceViewController as! TimeSetViewController
            
            let dateString = viewControllerTwo.dateSet
            
            self.destinationLabel.text = dateString
        }
        
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        dates()
    }
    
    func dates() {
        let presentDate = NSDate()
        
        let dateFormatter = NSDateFormatter()
        
        dateFormatter.dateFormat = "MMM-dd-yyyy"
        
        let todayString = dateFormatter.stringFromDate(presentDate)
        presentTimeLabel.text = todayString
        
    }


    @IBAction func warpButton(sender: UIButton) {
    
        timeUntilWarpLabel.text = "\(count)"
        
        timer = NSTimer.scheduledTimerWithTimeInterval(speed,
            target: self,
            selector: "updateUI",
            userInfo: nil,
            repeats: true)


        }
    func updateUI() {
        
        count = count - 1
        
        if count > 0 {
            
            timeUntilWarpLabel.text = "\(count)"
        
        } else {
            timeUntilWarpLabel.text = "WARP"
            
            timer?.invalidate()
            
            self.performSegueWithIdentifier("warpSegue", sender: self)
        
        }
    }
        @IBAction func warpTimeSegue(segue: UIStoryboardSegue) {
            
            if segue.identifier == "warpSegue" {
                
                lastTimeLabel.text = presentTimeLabel.text
                
                presentTimeLabel.text = destinationLabel.text
                
                count = 10
                
                timeUntilWarpLabel.text = "--"
                
            }
            
    }
}

